### ISSUE TITLE
A font that doesn't work in 20.1 but works in 19.5 (Android)

### ISSUE THREAD

**Issue by AlyafeiAli** [xbmc/xbmc#23300]
### Describe the bug
Here is a clear and concise description of what the problem is:  

In Android Kodi 19, I was able to use this ttf font and override the embedded sub's font. However, in Android Kodi 20.1 (Fresh install) it doesn't work at all and it uses the default font instead. I tried restarting many times but it still doens't work, and then I tried going back to Kodi 19.5 and the font started working again.  
**Kindly note that this issue is not present in Windows Kodi 20.1. The font works perfectly there**

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

  
The ttf font should work in Android Kodi 20.1 like it used to work in Android Kodi 19 and the Windows version of Kodi 20.1.  
The following screenshot is from Windows version which works fine:  
![Screenshot (1143)](https://github.com/xbmc/xbmc/assets/52626961/c07477be-9e9e-4d81-a0d5-8d8452ad0414)

## Actual Behavior

  
Unlike Android Kodi 19 and Windows Kodi 20.1, the font doesn't work in Android Kodi 20.1.  
The following screenshot is from Android Kodi 20.1:  
![image](https://github.com/xbmc/xbmc/assets/52626961/d9486da8-5942-439f-9194-9a675e3d7b9d)

### To Reproduce

1. (Android ) Put font file .ttf inside /.kodi/media/Fonts
2. Choose the font from subtitle settings
3. Enable override ASS/SSA subtitles fonts
4. Change the size settings and position settings
5. Play any mkv file with embedded ASS subs
6. You'll notice that the size settings and position settings have been applied, but not the fonts settings.

### Debuglog

The debuglog can be found here:  
https://paste.kodi.tv/qepedipazo.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

  
This is the ttf font file in question:  
[Bahij Jalal-Regular Edit.zip](https://github.com/xbmc/xbmc/files/11510454/Bahij.Jalal-Regular.Edit.zip)

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Android 11 and Android 12 ( Galaxy S10e and Shield TV)
 - Kodi version:  
 - 20.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by scott967** [comment]
I forgot most of what I knew about libass since it was first implemented, but I think the key in your log is:
```
CDVDSubtitlesLibass: [ass] fontselect: Using default font family: (Bahij Jalal, 400, 0) -> ArialMT, 0, ArialMT
CDVDSubtitlesLibass: [ass] fontselect: (Bahij Jalal, 400, 0) -> ArialMT, 0, ArialMT
```


Also libass on Windows uses DirectWrite for font management, while Android and Linux use fontconfig so I suspect there is some difference there in run time performance.  What are the styles in the ass subs?

**Comment by AlyafeiAli** [comment]
I figured it would be more convenient for you to have the actual sub files, you can download them from here: https://subscene.com/subtitles/hunter-x-hunter-first-season/arabic/2223710  
In this specific case, I was playing ep 137

**Comment by CastagnaIT** [comment]
looking at styles the font name dont match and more likely this is the cause of problems

The font attached have family name of:
"Bahij Jalal Edit"

in styles of ep137 there is no this name (in the second field value):
```
Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
Style: Default,SKR HEAD1,120,&H00FFFFFF,&H000000FF,&H00233715,&H00000000,0,0,0,0,100,100,0,0,1,1.5,0.75,2,30,30,30,1
Style: Secondary,SKR HEAD1,120,&H00FFFFFF,&H000000FF,&H001F455B,&H00000000,0,0,0,0,100,100,0,0,1,1.5,0.75,2,30,30,30,1
Style: Default - italics,SKR HEAD1,120,&H00FFFFFF,&H000000FF,&H00233715,&H00000000,0,-1,0,0,100,100,0,0,1,1.5,0.75,2,30,30,30,1
Style: Default up,SKR HEAD1,105,&H00FFFFFF,&H000000FF,&H00233715,&H00000000,0,0,0,0,100,100,0,0,1,1.5,0.75,8,30,30,0,1
Style: شرح المفردات,AGA Abasan Regular,75,&H00FFFFFF,&H000000FF,&H0016A1F1,&H00000000,0,0,0,0,100,100,0,0,1,1.5,0.75,8,53,53,15,1
Style: Sign #1,AF_Jizan,75,&H00FEFDFC,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,4.5,0,2,83,83,81,1
Style: Title,AF_Ed Dammam,225,&H00000000,&H000000FF,&H00233715,&H00000000,0,0,0,0,100,100,0,0,1,0,0,2,45,45,45,1
Style: Sign,ACS  Yaqout Extra Bold,98,&H00382822,&H000000FF,&H0087B2B2,&H96000000,0,0,0,0,100,100,0,0,1,0,0,5,75,75,0,178
Style: a,Arial,53,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,-1,0,0,0,100,100,0,0,1,1.5,1.5,2,15,15,15,1
Style: Captions,Baar Sophia,72,&H00FFFFFF,&H000000FF,&H00000000,&H00685553,-1,0,0,0,100,100,0,0,1,4.5,4.5,8,120,120,54,0
Style: SignA,Baar Sophia,54,&H00434544,&H000000FF,&H00FFFFFF,&H00685553,-1,0,0,0,100,100,0,0,1,4.5,0,7,90,62,54,0
Style: SignB,Baar Sophia,80,&H00FFFFFF,&H000000FF,&H00434544,&H00685553,-1,0,0,0,100,100,0,0,1,4.5,0,8,120,120,54,0
Style: Heart,Baar Sophia,80,&H00685EAB,&H000000FF,&H00363B3B,&H00685553,-1,0,0,0,100,100,0,0,1,4.5,0,8,120,120,54,0
Style: Default - Margin,SKR HEAD1,120,&H00FFFFFF,&H000000FF,&H00233715,&H00000000,0,0,0,0,100,100,0,0,1,1.5,0.75,2,30,30,0,1
Style: ED6,Far.Two Light,75,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,-1,0,0,0,100,100,0,0,1,2.25,0.75,3,15,15,15,1
```
imo you should download right fonts specified in styles

**Comment by AlyafeiAli** [comment]
I think you missed the whole point, @CastagnaIT . I’m using ‘Enable override ASS/SSA subtitles fonts’ which works as expected in kodi 19.5 but not in kodi 20.1

**Comment by CastagnaIT** [comment]
yes i did not read the word "override" above.. then need to test it

**Comment by CastagnaIT** [comment]
found this for now, but need more investigations

seem your font have font names stored with different names

on Windows result:
![immagine](https://github.com/xbmc/xbmc/assets/3257156/993548fd-1867-423a-82de-cf37b486a7bc)
on linux and android result:
![immagine](https://github.com/xbmc/xbmc/assets/3257156/786dad21-6de7-4430-bba7-9bb96b9646a3)
this could be one of the reasons why this font works only on Windows
due to different name its not recognized on other os's

i suspect is due to "preferred family" (famiglia preferita) field that its different
![immagine](https://github.com/xbmc/xbmc/assets/3257156/a0944f56-40a9-4dee-bc54-cbb1b0bf00d9)

**Comment by AlyafeiAli** [comment]
Thank you for investigating this issue, @CastagnaIT. Interestingly enough, the font works fine on the 20.1 Windows version.  
I hope this info helps you in a way.

**Comment by CastagnaIT** [comment]
mmh i have check better, and there is different family name on each platform,
where usually should be equal:
microsoft platform: "Bahij Jalal Edit"
macintosh platform: "Bahij Jalal"
now i dont have deeper knowledge on font structure but seem wrong

since on #23429 i prioritized the family name from microsoft, should "fix" the problem with this case
i will make a partial backport and i will provide you a test apk

**Comment by CastagnaIT** [comment]
Test builds:
http://mirrors.kodi.tv/test-builds/android/arm64-v8a/kodi-20230626-f6e1781b-fix_ttf_familyname-arm64-v8a.apk
http://mirrors.kodi.tv/test-builds/android/arm/kodi-20230626-f6e1781b-fix_ttf_familyname-armeabi-v7a.apk

you can try upgrade, but if fails you will have to remove the existing kodi 20

**Comment by AlyafeiAli** [comment]
Thank you!   
I'll test them asap and report back

**Comment by AlyafeiAli** [comment]
Good new!  
The issue is resolved. Thank you so much!

**Comment by gzn25** [comment]
I apologize for replying to this old issue.
I recently upgraded my Kodi from 18.9 to 21.0 on macOS and noticed that the ttc font in the ass subtitles is not displaying correctly anymore. It keeps fallback to the default font. When I change the font in subtitles to ttf, it displays correctly.
The font I'm using is STHeiti Medium.ttc. The same subtitles can be played in IINA and Infuse with no problem, and also fine in Infuse on iOS after I installed the font on the device.
